### PR TITLE
Add envSecrets block to HowTo Custom Provider tutorial

### PR DESCRIPTION
**Pull request by lakshmimsft**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.io/community/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
Add envSecrets block to HowTo Custom Provider tutorial

### CODE REVIEW

File: docs/content/guides/recipes/terraform/howto-custom-provider/index.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Create a Bicep file `env.bicep` with the secretStore resource:
 
 `recipeConfig/terraform/providers` allows you to setup configurations for one or multiple Terraform Providers. For more information refer to the [Radius Environment schema]({{< ref environment-schema >}}) page.
 
-In your `env.bicep` file add an Environment resource, along with Recipe configuration which leverages properties from the previously defined secret store. In this example you're also passing in `host` as an environment variable to highlight use cases where, depending on provider configuration requirements, users can pass environment variables to the Terraform recipes runtime.
+In your `env.bicep` file add an Environment resource, along with Recipe configuration which leverages properties from the previously defined secret store. In this example you're also passing in `host` and `port` as environment variables to highlight use cases where, depending on provider configuration requirements, users can pass environment variables as plain text and as secret values to the Terraform recipes runtime.
```
Comment (willtsai) [inline docs/content/guides/recipes/terraform/howto-custom-provider/index.md]: would this be more accurate/clear?
```suggestion
In your `env.bicep` file add an Environment resource, along with Recipe configuration which leverages properties from the previously defined secret store. In this example you're also passing in `host` and `port` as environment variables to highlight use cases where, depending on provider configuration requirements, users can pass environment variables as plain text and as `envSecrets` values to the Terraform recipes runtime.
```

**Review by willtsai — DISMISSED** [review]
I like the addition, just one minor suggestion for clarity

File: docs/content/guides/recipes/terraform/howto-custom-provider/snippets/env-complete.bicep [hunk 2]
```diff
@@ -55,7 +57,13 @@ resource env 'Applications.Core/environments@2023-10-01-preview' = {
         }
       }
       env: {
-        PGHOST: 'postgres.corerp-resources-terraform-pg-app.svc.cluster.local'
+        PGPORT: '5432'
+      }
+      envSecrets: {
+        PGHOST: {
+          source: pgsSecretStore.id
+          key: 'host'
```
Comment (willtsai) [inline docs/content/guides/recipes/terraform/howto-custom-provider/snippets/env-complete.bicep:65]: I think you also need to add the `{ host: myhost }` entry into the `pgsSecretStore`?

Comment (lakshmimsft) [reply docs/content/guides/recipes/terraform/howto-custom-provider/snippets/env-complete.bicep:65]: ln 24-26

Comment (willtsai) [reply docs/content/guides/recipes/terraform/howto-custom-provider/snippets/env-complete.bicep:65]: sorry I missed this! pls disregard and i'll resolve this comment

File: docs/content/guides/recipes/terraform/howto-custom-provider/index.md (hunk 1 above)
Comment (lakshmimsft) [reply docs/content/guides/recipes/terraform/howto-custom-provider/index.md]: updated to `secret values in 'envSecrets' block` as discussed

**Review by willtsai — APPROVED** [review]

**Pull request merged**